src/audio/copier/copier.c

```diff
@@ -740,6 +740,29 @@ static struct comp_dev *copier_new(const struct comp_driver *drv,
 	}
 
 	dev->pipeline = ipc_pipe->pipeline;
+	/* Calculation of the period in which the pipeline should be scheduled based on
+	 * the input buffer size.
+	 */
+	if (copier->gtw_cfg.dma_buffer_size > 0) {
+		/* Size of the single sample in bytes. */
+		uint32_t sample_size = copier->base.audio_fmt.channels_count *
+				       (copier->base.audio_fmt.depth / 8);
+		/* Size of the one millisecond of audio data in bytes. */
+		uint32_t one_ms = (sample_size * copier->base.audio_fmt.sampling_frequency) / 1000;
+		/* The number of milliseconds of data that will fit into the buffer. */
+		uint32_t ms_count = copier->gtw_cfg.dma_buffer_size / one_ms;
```

Comment:
if there is reminder, then ms_count is not accurate, possible add this?

if (copier->gtw_cfg.dma_buffer_size % one_ms)
           ms_count++;

Reply:
if it's needed, then `SOF_DIV_ROUND_UP()` should be used